env.d.ts

```diff
@@ -1,7 +1,23 @@
+import 'vue-router'
 /// <reference types="vite/client" />
 declare module '*.vue' {
   import type { DefineComponent } from 'vue'
   // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/ban-types
   const component: DefineComponent<{}, {}, any>
   export default component
 }
+declare global {
+  import type { SmartContractRequestParams } from '@/utils/signerMakeTx'
+
+  interface Window {
+    signerMakeTx: (data: SmartContractRequestParams) => Promise<string>
+  }
+}
+
+declare module 'vue-router' {
+  interface RouteMeta {
+    requiresAuth: boolean
```

Comment:
Let's keep this field optional.

```suggestion
    requiresAuth?: boolean
```